### DISCUSSION TITLE
The OnClick event runs on Page Load (App Directory)

### DISCUSSION THREAD

**Discussion by DanavsSite** [vercel/next.js#45672] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

The OnClick Event runs both when the page is loaded and the button is clicked i tried to fix it by using arrow functions but it didnt work
code :
```js
import db from "@/lib/db";

export default async function Page() {

  const data = await db.project.findMany({...});
  const crUser = async () => {
    await db.project.create({...});
  };
  return (
    <>
      <div className="grid align-middle">
        <button
          onClick={await crUser()}
          className="..."
        >
          <span
            style={{...}}
            className="..."
          >
            + Create A Project
          </span>
        </button>
      </div>
    </>
  );
}

```

### Link to the code that reproduces this issue

a button which runs a async function in app directory

### To Reproduce

```js
import db from "@/lib/db";

export default async function Page() {

  const data = await db.project.findMany({...});
  const crUser = async () => {
    await db.project.create({...});
  };
  return (
    <>
      <div className="grid align-middle">
        <button
          onClick={await crUser()}
          className="..."
        >
          <span
            style={{...}}
            className="..."
          >
            + Create A Project
          </span>
        </button>
      </div>
    </>
  );
}

```

### Describe the Bug

The OnClick Function runs on the page load as well as when the button is clicked and also on reload i tried to fix it by using 
```async () => await function() ``` but it didnt work

### Expected Behavior

Only run the function on click

### Which browser are you using? (if relevant)

Opera GX

### How are you deploying your application? (if relevant)

next dev

**Comment by clicktodev** [comment]
You need to pass a reference of the function but instead you're invoking it.
Change it to this `onClick={crUser}`
or this `onClick={() => crUser()}`
